histomicsui/web_client/panels/DrawWidget.js

```diff
@@ -170,6 +192,73 @@ var DrawWidget = Panel.extend({
         });
     },
 
+    /**
+     * Respond to a click on the "view" button by changing the
+     * viewer location and zoom level to focus on one annotation
+     */
+    viewElement(evt) {
+        const annot = this.collection._byId[$(evt.target).parent().attr('data-id')];
+        let points;
+        let pointAnnot = false;
+        switch (annot.get('type')) {
+            case 'point':
+                points = [annot.get('center')];
+                pointAnnot = true;
+                break;
+            case 'polyline':
+                points = annot.get('points');
+                break;
+            case 'rectangle':
+                points = convertRectangle(annot.attributes).coordinates[0];
+                break;
+            case 'ellipse':
+                points = convertEllipse(annot.attributes).coordinates[0];
+                break;
+            case 'circle':
+                points = convertCircle(annot.attributes).coordinates[0];
+                break;
+        }
+        const xCoords = points.map((point) => {
+            return point[0];
+        });
+        const yCoords = points.map((point) => {
+            return point[1];
+        });
+        const bounds = {
+            left: Math.min(...xCoords),
+            top: Math.min(...yCoords),
+            right: Math.max(...xCoords),
+            bottom: Math.max(...yCoords)
+        };
+        const map = this.parentView.viewer;
+        const originalZoomRange = map.zoomRange();
+        map.zoomRange({
+            min: Number.NEGATIVE_INFINITY,
+            max: Number.POSITIVE_INFINITY
+        });
+        const newView = pointAnnot ?
+                        {
+                            center: {
+                                x: bounds.left,
+                                y: bounds.top
+                        },
+                        zoom: false
+                        } :
+                        map.zoomAndCenterFromBounds(bounds, map.rotation());
+        map.zoomRange(originalZoomRange);
+        if (Math.abs(this.zoomWidget.zoomToMagnification(newView.zoom) * 0.2968 + 1.8666 - this.zoomWidget.zoomToMagnification(map.zoom())) <= 40 && map.zoom() < newView.zoom) {
+            newView.zoom = false;
+        }
+        map.transition({
+            center: newView.center,
+            duration: ((newView.center.x - map.center().x) ** 2 + (newView.center.y - map.center().y) ** 2) ** 0.5,
```

Comment:
I originally started with a fixed time, but I thought that there were times when the transition between annotations that were far apart was too fast, so I tried to implement something that was responsive to how far apart the two annotations were. But if you think it's too slow I'll just go back to the fixed time or put an upper limit on how long it can take.

As for a different animation, the `d3.interpolateZoom` is what Mary originally suggested and what I tried out at first, but what I found was that on smaller images it sometimes looked jittery.  Also it doesn't look very good when it tries to go to an annotation at the edge of the image when clamping bounds are on (this can also be an issue with the linear animation, but it's more extreme with the interpolate zoom).  Here are a couple of examples (sorry they're so pixelated, I ran into GitHub's file limit):
![Screencast_from_06-22-2022_11_02_30_AM_AdobeExpress (1)](https://user-images.githubusercontent.com/87092433/175068286-f0d9e6aa-dfdd-4120-9676-4499675f0088.gif)
![Screencast_from_06-22-2022_11_00_40_AM_AdobeExpress](https://user-images.githubusercontent.com/87092433/175069077-790bfe54-3e90-454f-aca6-4f66578507fc.gif)

Reply:
We're often working with images that are huge (100k square), so using the distance as the time ends up being much to slow -- maybe using the distance metric with an upper limit or a power function would be better.

Reply:
I added a lower and upper limit to the duration of the transition and used `d3.easeExpInOut` to accomplish the acceleration and deceleration of the camera as the transition occurred.

See e7d303a